agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/AlleleIndexerRepository.java

```diff
@@ -135,4 +138,12 @@ public AlleleDocumentCache getAlleleDocumentCache(String species) {
         return getMapSetForQuery(query, "a.primaryKey", "term.name", getSpeciesParams(species));
     }
 
+    public Map<String, Set<String>> getMolecularConsequence(String species) {
+        String query = "MATCH (species:Species)--(:Gene)-[:IS_ALLELE_OF]-(a:Allele)--(v:Variant)--(consequence:GeneLevelConsequence) ";
```

Comment:
Depending on the speed of this query you might want to put all the relationships in it will help speed it up if there are issues.